contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -876,6 +876,20 @@ kind `Foo` by just the name (within the current namespace, if a namespaced
 resource), or should be called `fooRef`, and should contain a subset of the
 fields of the `ObjectReference` type.
 
+Object references on a namespaced type should usually refer only to objects in
+the same namespace.  Because namespaces are a security boundary, cross namespace
+references can have unexpected impacts, including:
+ 1. leaking information about one namespace into another namespace. It's natural to place status messages or even bits of
+    content about the referenced object in the original. This is a problem across namespaces.
+ 2. potential invasions into other namespaces. Often references give access to a piece of referred information, so being
+    able to express "give me that one over there" is dangerous across namespaces without addition work for permission checks
+    or opt-in's from both involved namespaces.
+ 3. referential integrity problems that one party cannot solve. Referencing namespace/B from namespace/A doesn't imply the
+    power to control the other namespace. This means that you can refer to a thing you cannot create or update.
```

Comment:
> There's no guarantee that the author of the `CoolSource` has RBAC access to view or otherwise interact with a `CoolTarget` within the same namespace. That said, it does _seem_ less likely (but obviously not impossible) that the author would have any kind of RBAC access to another namespace.

There is no guarantee, but it is highly likely that inside of a single namespace the resources are at least viewable and likely edit-able.